Simplenote/SharedStorageMigrator.swift

```diff
@@ -0,0 +1,154 @@
+import Foundation
+import CoreData
+
+class SharedStorageMigrator {
+
+    static let documentsURL: URL? = {
+        FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first
+    }()
+
+    static let groupDirectory: URL? = {
+        FileManager.default.containerURL(forSecurityApplicationGroupIdentifier: Constants.sharedDirectoryDomain + Constants.groupIdentifier)
+    }()
+
+    static let groupDocumentsDirectory: URL? = {
+        guard let groupDirectory = groupDirectory else {
+            return nil
+        }
+        return groupDirectory.appendingPathComponent(Constants.documentDirectory)
+    }()
+
+    static let oldDbURL: URL? = {
+        guard let documentsURL = documentsURL else {
+            return nil
+        }
+
+        return documentsURL.appendingPathComponent(Constants.sqlFile)
+    }()
+
+    static let newDbURL: URL? = {
+        guard let groupDocumentsDirectory = groupDocumentsDirectory else {
+            return nil
+        }
+        return groupDocumentsDirectory.appendingPathComponent(Constants.sqlFile)
+    }()
+
+    /// Database Migration
+    /// To be able to share data with app extensions, the CoreData database needs to be migrated to an app group
+    /// Must run before Simperium is setup
+    static func migrateCoreDataToAppGroup() {
+        guard let oldDbURL = oldDbURL,
+              let groupDocumentsDirectory = groupDocumentsDirectory,
+              let newDbURL = newDbURL else {
+            return
+        }
+
+        // Testing prints
+        // TODO: Remove prints later
+        print("oldDb exists \(FileManager.default.fileExists(atPath: oldDbURL.path))")
+        print(oldDbURL.path)
+        print("newDb exists \(FileManager.default.fileExists(atPath: newDbURL.path))")
+        print(newDbURL.path)
+
+        let persistentStoreCoordinator = NSPersistentStoreCoordinator(managedObjectModel: SPAppDelegate.shared().managedObjectModel)
+
+        if FileManager.default.fileExists(atPath: newDbURL.path) {
+            // Migration previously completed
+            NSLog("Database migration not needed")
+
+            // exit
+            return
+        }
+
+        if !FileManager.default.fileExists(atPath: oldDbURL.path) && !FileManager.default.fileExists(atPath: newDbURL.path) {
```

Comment:
This flow should **not** be required IMO. Bear in mind that Simperium will be in charge of spawning the main PSC.

We'd probably need, though, a clean way to determine the Storage URL (that is: by avoiding code duplication).